MainSelectorComponent.py

```diff
@@ -267,106 +242,86 @@ def update(self):
-			self._update_control_channels()
-			self._mode_index = 5
 		elif mode == "user 2":
-			self._setup_session(not as_active, not as_enabled)
-			self._setup_instrument_controller(not as_active)
-			self._setup_device_controller(not as_active)
-			self._setup_mixer(not as_active)
-			self._setup_step_sequencer(not as_active)
-			self._setup_step_sequencer2(not as_active)
+			self._control_surface.show_message("USER 2 MODE" )
 			self._setup_user_mode(False, False, False, False)
-			self._update_control_channels()
 			self._mode_index = 2
+		elif mode == "mixer":
+			self._control_surface.show_message("MIXER MODE")
+			self._setup_session(False, True)
+			self._setup_mixer(True)
+			self._mode_index = 3
+		
+		self._update_control_channels()#change channel for all buttons according to mode
+		
+		if(self._mode_index in range(1,3)):
 			self._osd.clear()
-			self._osd.mode = "User 2"
+			self._osd.mode = mode.capitalize()
 			self._osd.update()
-		
-	def _setup_session(self, as_active, as_enabled):
-		assert isinstance(as_active, type(False))
+			
+	def _setup_session(self, as_active, as_navigation_enabled):
+		assert isinstance(as_active, type(False))#assert is boolean
 		for button in self._nav_buttons:
-			if as_enabled:
+			if as_navigation_enabled:
 				button.set_on_off_values("Mode.Session.On", "Mode.Session.Off")
 			else:
 				button.set_on_off_values("DefaultButton.Disabled", "DefaultButton.Disabled")
 
-		# matrix
 		self._activate_matrix(True)
-		for scene_index in range(self._session._num_scenes):
+		for scene_index in range(self._session._num_scenes):#iterate over scenes
 			scene = self._session.scene(scene_index)
-			if as_active:
+			if as_active:#set scene launch buttons
 				scene_button = self._side_buttons[scene_index]
 				scene_button.set_enabled(as_active)
 				scene_button.set_on_off_values("DefaultButton.Disabled", "DefaultButton.Disabled")
 				scene.set_launch_button(scene_button)
 			else:
 				scene.set_launch_button(None)  
-			for track_index in range(self._session._num_tracks):
-				if as_active:
+				
+			for track_index in range(self._session._num_tracks):#iterate over tracks of a scene -> clip slots
+				if as_active:#set clip slot launch button
 					button = self._matrix.get_button(track_index, scene_index)
 					button.set_on_off_values("DefaultButton.Disabled", "DefaultButton.Disabled")
 					button.set_enabled(as_active)
 					scene.clip_slot(track_index).set_launch_button(button)
 				else:
 					scene.clip_slot(track_index).set_launch_button(None)
 
-		if as_active:
+		if as_active:#Set up stop clip buttons and stop all clips button
 			if self._session._stop_clip_buttons != None:
 				for button in self._session._stop_clip_buttons:
 					button.set_enabled(as_active)
 				#	button.set_on_off_values("Session.StopClip", "DefaultButton.Disabled")
 				self._session.set_stop_track_clip_buttons(self._session._stop_clip_buttons)
-
+				
+				#last side button is stop all clip buttons ???
```

Comment:
yes last scene button is a stop all, if last row of the matrix is acting as stop buttons